amqp/readers_writers.go

```diff
@@ -176,11 +184,14 @@ func ReadShortstr(r io.Reader) (data string, err error) {
 
 // WriteShortstr writes string
 func WriteShortstr(wr io.Writer, data string) error {
-	err := binary.Write(wr, binary.BigEndian, byte(len(data)))
-	if err != nil {
+	if err := WriteOctet(wr, byte(len(data))); err != nil {
```

Comment:
What happens if `len(data)` > 255?

Reply:
Will be strange behavior
yes, need to fix that with comparisions and throw errors